### ISSUE TITLE
Compilation error on Amazon Linux AMI.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#463]
Compilation on the Amazon Linux AMI currently fails with errors of the form (this is probably the same on CentOS).

```
In file included from /home/ec2-user/ray/src/common/state/table.h:7:0,
                 from /home/ec2-user/ray/src/common/state/table.cc:1:
/home/ec2-user/ray/src/common/state/table.cc: In function ‘int64_t table_timeout_handler(event_loop*, int64_t, void*)’:
/home/ec2-user/ray/src/common/state/table.cc:90:49: error: expected ‘)’ before PRId64’
     LOG_WARN("Table command %s with timer ID %" PRId64 " failed",
                                                 ^
/home/ec2-user/ray/src/common/cmake/../common.h:75:37: note: in definition of macro ‘LOG_WARN’
   fprintf(stderr, "[WARN] (%s:%d) " M "\n", __FILE__, __LINE__, ##__VA_ARGS__)
                                     ^
```

Replacing the instances of PRId64 with `%lld` or something like that (or just removing them) makes compilation work, although that probably isn't exactly the right solution (because it will introduce compiler warnings at least on Ubuntu).

Note that to install the Ray dependencies, I did the following.

```bash
sudo yum update
sudo yum install -y git cmake autoconf curl libtool boost-devel boost-devel-static unzip
sudo yum install -y gcc gcc-c++ make openssl-devel

wget https://repo.continuum.io/archive/Anaconda3-4.3.1-Linux-x86_64.sh
bash Anaconda3-4.3.1-Linux-x86_64.sh
pip install numpy cloudpickle funcsigs colorama psutil redis flatbuffers
```

**Comment by atumanov** [comment]
this should fix it:
```
diff --git a/src/common/common.h b/src/common/common.h
index cd75148..dc15a42 100644
--- a/src/common/common.h
+++ b/src/common/common.h
@@ -6,6 +6,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
 #include <inttypes.h>
 #ifndef _WIN32
 #include <execinfo.h>
diff --git a/src/plasma/plasma.h b/src/plasma/plasma.h
index 5329789..7310ef0 100644
--- a/src/plasma/plasma.h
+++ b/src/plasma/plasma.h
@@ -1,7 +1,6 @@
 #ifndef PLASMA_H
 #define PLASMA_H
 
-#include <inttypes.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <errno.h>
@@ -11,6 +10,7 @@
 #include <unistd.h> /* pid_t */
 
 #include "common.h"
+#include <inttypes.h>
 #include "format/common_generated.h"
 
 #include "utarray.h"
```

**Comment by atumanov** [comment]
p.s. inttypes.h needs to be preceded by #defining these STDC macros. That's why it's moved in plasma.h. An alternative is to #include common.h higher up or just #define the macros again.

**Comment by atumanov** [comment]
there's a commit for this fix here : e4933aafb52a66ce04e473d757dd4b824f188745

**Comment by robertnishihara** [comment]
This should be addressed by #472.

**Comment by robertnishihara** [comment]
Reopening because #579 seems to have broken compilation on CentOS. I wonder if we should really be testing this in Travis. Otherwise, it's unclear how we'll avoid more occurrences of this problem in the future.

Compilation on CentOS gives the following error.

```
/ray/src/plasma/plasma_store.cc: In function ‘int main(int, char**)’:
/ray/src/plasma/plasma_store.cc:638:40: error: expected ‘)’ before ‘SCNd64’
       int scanned = sscanf(optarg, "%" SCNd64 "%c", &system_memory, &extra);
                                        ^
```

**Comment by robertnishihara** [comment]
Closing because this is fixed by #622, but we really should test CentOS in Travis as described in #625.
